File: Library.Core.Web/src/app/view-models/publication-house/putPublication-houseView.ts
export class PutPublicationHouseView {
  constructor(
    public id: number = 0,
    public name?: string,
    public adress?: string,
    public books?: BookPutPublicationHouseViewItem[]
  ) { }
}

export class BookPutPublicationHouseViewItem {
  constructor(
    public id: number = 0,
    public name?: string,
    public yearOfPublishing: number = 0
  ) { }
}

File: Library.Core.ViewModelLayer/ViewModels/PublicationHouseViewModels/PutPublicationHouseView.cs
using System.Collections.Generic;

namespace Library.Core.ViewModelLayer.ViewModels.PublicationHouse
{
    public class PutPublicationHouseView
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Adress { get; set; }

        public virtual ICollection<BookPutPublicationHouseViewItem> Books { get; set; }

        public PutPublicationHouseView()
        {
            Books = new List<BookPutPublicationHouseViewItem>();
        }
    }

    public class BookPutPublicationHouseViewItem
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int YearOfPublishing { get; set; }

        public BookPutPublicationHouseViewItem()
        {

        }
    }
}
